packages/flutter_test/lib/src/controller.dart

```diff
@@ -23,6 +23,144 @@ const double kDragSlopDefault = 20.0;
 
 const String _defaultPlatform = kIsWeb ? 'web' : 'android';
 
+/// Class that programatically interacts with the Semantics tree.
+class SemanticsController {
+  /// Creates a SemanticsController that uses the given binding.
+  SemanticsController(this.binding);
+
+  static final int _scrollingActions =
+    SemanticsAction.scrollUp.index |
+    SemanticsAction.scrollDown.index |
+    SemanticsAction.scrollLeft.index |
+    SemanticsAction.scrollRight.index;
+
+  static final int _importantFlags =
+    SemanticsFlag.hasCheckedState.index |
+    SemanticsFlag.hasToggledState.index |
+    SemanticsFlag.hasEnabledState.index |
+    SemanticsFlag.isButton.index |
+    SemanticsFlag.isTextField.index |
+    SemanticsFlag.isFocusable.index |
+    SemanticsFlag.isSlider.index |
+    SemanticsFlag.isInMutuallyExclusiveGroup.index;
+
+
+  /// A reference to the current instance of the binding.
+  final WidgetsBinding binding;
```

Comment:
Good call. I was copying `WidgetController`, but since the `SemanticsController` should always be accessed through `WidgetController` anyways, and since the `binding` is the same for both, no reason for this _not_ to be private.